Detectors/MUON/MID/Workflow/src/FilteringBCSpec.cxx

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file   MID/Workflow/src/FilteringBCSpec.cxx
+/// \brief  MID filtering spec
+/// \author Diego Stocco <Diego.Stocco at cern.ch>
+/// \date   16 March 2022
+
+#include "MIDWorkflow/FilteringBCSpec.h"
+
+#include <vector>
+#include <gsl/gsl>
+#include <fmt/format.h>
+#include "Framework/CCDBParamSpec.h"
+#include "Framework/Output.h"
+#include "Framework/Task.h"
+#include "Framework/WorkflowSpec.h"
+#include "DataFormatsMID/ColumnData.h"
+#include "DataFormatsMID/ROFRecord.h"
+#include "DataFormatsMID/MCLabel.h"
+#include "DetectorsBase/GRPGeomHelper.h"
+#include "SimulationDataFormat/MCTruthContainer.h"
+#include "MIDFiltering/FiltererBC.h"
+#include "MIDFiltering/FiltererBCParam.h"
+#include "MIDSimulation/DigitsMerger.h"
+#include "MIDWorkflow/ColumnDataSpecsUtils.h"
+
+namespace of = o2::framework;
+
+namespace o2
+{
+namespace mid
+{
+
+class FilteringBCDeviceDPL
+{
+ public:
+  FilteringBCDeviceDPL(std::shared_ptr<o2::base::GRPGeomRequest> gr, bool useMC, std::vector<of::OutputSpec> outputSpecs) : mGGCCDBRequest(gr), mUseMC(useMC)
+  {
+    mOutputs = specs::buildOutputs(outputSpecs);
+  }
+
+  void init(of::InitContext& ic)
+  {
+    o2::base::GRPGeomHelper::instance().setRequest(mGGCCDBRequest);
+    mFiltererBC.setBCDiffLow(FiltererBCParam::Instance().maxBCDiffLow);
+    mFiltererBC.setBCDiffHigh(FiltererBCParam::Instance().maxBCDiffHigh);
+    mFiltererBC.setSelectOnly(FiltererBCParam::Instance().selectOnly);
+  }
+
+  void finaliseCCDB(of::ConcreteDataMatcher matcher, void* obj)
+  {
+    if (o2::base::GRPGeomHelper::instance().finaliseCCDB(matcher, obj)) {
+      LOG(info) << "Update Bunch Filling for MID";
+      mFiltererBC.setBunchFilling(o2::base::GRPGeomHelper::instance().getGRPLHCIF()->getBunchFilling());
+      return;
+    }
+  }
+
+  void run(of::ProcessingContext& pc)
+  {
+    updateTimeDependentParams(pc);
+
+    auto data = specs::getData(pc, "mid_filter_BC_in", EventType::Standard);
+    auto inROFRecords = specs::getRofs(pc, "mid_filter_BC_in", EventType::Standard);
+
+    std::unique_ptr<const o2::dataformats::MCTruthContainer<MCLabel>> inMCContainer = mUseMC ? specs::getLabels(pc, "mid_filter_BC_in") : nullptr;
```

Comment:
any particular reason you're not using the simpler `auto inMCContainer = ...` like for the other inputs ?